src/lib/retire-js/RetireJsEngine.ts

```diff
@@ -245,45 +258,120 @@ export class RetireJsEngine implements RuleEngine {
 		return await this.config.isEngineEnabled(RetireJsEngine.ENGINE_ENUM);
 	}
 
-	private async createPathAlias(tmpParent: string, basePath: string): Promise<string> {
-		// We want to make sure each duplicate file's path is unique, to avoid collisions. If we haven't already created
-		// a unique alias for this path's directory, we need to do that now.
-		const baseDir = path.dirname(basePath);
-		if (!this.aliasesByOriginalPath.has(baseDir)) {
-			const aliasDir = path.join(tmpParent, `TMP_${RetireJsEngine.NEXT_ALIAS_IDX++}`);
-			this.aliasesByOriginalPath.set(baseDir, aliasDir);
-			this.originalPathsByAlias.set(aliasDir, baseDir);
-			return new Promise((res, rej) => {
-				// Create a directory beneath the temporary parent directory with the name of the alias.
-				fs.mkdir(aliasDir, (err) => {
-					if (err) {
-						rej(err);
-					} else {
-						res(aliasDir);
-					}
-				});
-			});
+	private getNextDirAlias(): string {
+		return `TMPDIR_${RetireJsEngine.NEXT_TMPDIR_IDX++}`;
+	}
+
+	private getNextFileAlias(): string {
+		return `TMPFILE_${RetireJsEngine.NEXT_TMPFILE_IDX++}.js`;
+	}
+
+	private async createAliasDirectory(tmpParent: string, originalPath: string): Promise<string> {
+		// By converting the directory portion of each path into a unique alias, we can make sure that the alias paths
+		// remain unique (and thereby avoid name collision) without renaming the files themselves.
+		const originalDir = path.dirname(originalPath);
+		if (!this.aliasDirsByOriginalDir.has(originalDir)) {
+			const aliasDir = path.join(tmpParent, this.getNextDirAlias());
+			this.aliasDirsByOriginalDir.set(originalDir, aliasDir);
+			await this.fh.mkdirIfNotExists(aliasDir);
+		}
+		return this.aliasDirsByOriginalDir.get(originalDir);
+	}
+
+
+
+	private async duplicateStaticResource(tmpParent: string, fileName: string): Promise<void> {
+		const resourceType: StaticResourceType = await this.srh.identifyStaticResourceType(fileName);
+
+		switch (resourceType) {
+			case StaticResourceType.JS:
+				return this.duplicateJsFile(tmpParent, fileName);
+			case StaticResourceType.ZIP:
+				return this.duplicateZipFile(tmpParent, fileName);
+			case StaticResourceType.OTHER:
+				this.logger.trace(`Static Resource ${fileName} is neither JS nor ZIP. It is being skipped.`);
+				return;
+			case null:
```

Comment:
Could this be changed to `default:`? if not can, you add a `default` case? it's best for switch statements to have a default, even if it just throws an exception because it should never happen.